### DISCUSSION TITLE
Can GRUB, GRUB2 boot Windows via WIMBOOT?

### DISCUSSION THREAD

**Discussion by JunielKatarn** [ipxe/ipxe#481] [Q&A] [answered]
Context:
- Disabling Secure Boot is **not an option** in my case, so I can't use iPXE.efi at all.
- Supposedly, WimBoot 2.7.2 is signed and usable in Secure Boot. https://twitter.com/ipxe/status/1364578824558698500

Is there a way to either:
1. Use a Linux, SB-signed GRUB2 instance to "invoke" WimBoot and boot Windows accordingly?
2. Use WimBoot as a first EFI executable and configure the Windows Boot via a configuration file?

**Comment by NiKiZe** [comment, marked as answer]
I think this is an issue you should take up with the grub mailing list.
I know that this has been asked, and answered several times in the past - but of course not able to find the precis post :/

wimboot is a efi64 binary, combined with a "linux kernel look-alike binary" which makes the same binary usable for both efi and pcbios.
It is also signed by Microsoft

Any efi loader could run this, the only thing you have to do is to load the needed files into memory so they are available in "efifs" and then execute wimboot as an efi binary. (I have not seen any way to do this in grub, mainly due to grub incorrectly running the "kernel" path rather then the "efi" path)
